resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -155,7 +155,7 @@ private[spark] class KubernetesClusterSchedulerBackend(
   private def allocateNewExecutorPod(): (String, Pod) = {
     val executorKubernetesId = UUID.randomUUID().toString.replaceAll("-", "")
     val executorId = EXECUTOR_ID_COUNTER.incrementAndGet().toString
-    val name = s"exec$executorKubernetesId"
+    val name = s"$kubernetesDriverServiceName-exec$executorKubernetesId"
```

Comment:
That is a good point. I'll check the constraints we impose and see if some truncation is needed. IIRC, it's the service names that typically complain about length because they're involved in DNS domain creation.